pkg/js/validator/validate-rules.ts

```diff
@@ -0,0 +1,46 @@
+export const Rules = {
+  type: "[^:#@\\s]{1,254}",
+  relation: "[^:#@\\s]{1,50}",
+  condition: "[^\\s]{2,256}",
+  object: "[^\\s]{2,256}",
+  id: "[^#:\\s]+",
+};
+
+export const Validate = {
+  // An Object is composed of a type and identifier (e.g. 'document:1')
+  object: (object: string): boolean => {
+    return validateFieldValue(`^${Rules.object}:${Rules.id}$`, object);
```

Comment:
```suggestion
    return validateFieldValue(`^${Rules.type}:${Rules.id}$`, object);
```

This should ALSO pass Rules.object, but Rules.object is for the full thing

Reply:
Is there a reason to keep `Rules.object` definition then? It appears to only be used to enforce string length, and that it includes no spaces.

Both `Rules.type` and `Rules.id` are more strict and enforce no spacing. Can I just add length check to the strings if that is the case?

Reply:
Also does the `id` have a max length, or is this just checked by `Rules.object`